gateway/app/main/processes/controller.py

```diff
@@ -111,17 +112,17 @@ def delete_cycle(self, **kwargs):
         """
         self._cycles.delete(**kwargs)
 
-    def last_participation(self, worker_id: str, model_id: str, version: str) -> int:
+    def last_participation(self, worker_id: str, name: str, version: str) -> int:
         """ Retrieve the last time the worker participated from this cycle.
             Args:
                 worker_id: Worker's ID.
-                model_id: Model's ID.
+                name: Federated Learning Process Name.
                 version: Model's version.
             Return:
                 last_participation: Index of the last cycle assigned to this worker.
         """
-        _model = self._models.first(id=model_id)
-        _cycles = self._cycles.query(fl_process_id=_model.fl_process_id)
+        _fl_process = self._processes.first(name=name)
```

Comment:
Is version ignored? (That's fine for now, can be addressed later)

Reply:
Done!